x-pack/legacy/plugins/monitoring/index.js

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get } from 'lodash';
+import { resolve } from 'path';
+import { config } from './config';
+import { deprecations } from './deprecations';
+import { getUiExports } from './ui_exports';
+import { KIBANA_ALERTING_ENABLED } from './common/constants';
+import { telemetryCollectionManager } from '../../../../src/legacy/core_plugins/telemetry/server';
+
+/**
+ * Invokes plugin modules to instantiate the Monitoring plugin for Kibana
+ * @param kibana {Object} Kibana plugin instance
+ * @return {Object} Monitoring UI Kibana plugin object
+ */
+const deps = ['kibana', 'elasticsearch', 'xpack_main'];
+if (KIBANA_ALERTING_ENABLED) {
+  deps.push(...['alerting', 'actions']);
+}
+export const monitoring = kibana => {
+  let legacyApi = {};
+  return new kibana.Plugin({
+    require: deps,
+    id: 'monitoring',
+    configPrefix: 'monitoring',
+    publicDir: resolve(__dirname, 'public'),
+    init(server) {
+      const serverConfig = server.config();
+      const xpackMain = server.plugins.xpack_main;
+      const elasticsearch = server.plugins.elasticsearch;
+      const getOSInfo = server.getOSInfo;
+      const hapiServer = server;
+      const kbnServerStatus = this.kbnServer.status;
+      legacyApi = {
+        xpackMain,
+        telemetryCollectionManager,
+        elasticsearch,
```

Comment:
don't think you need to pass it to NP

Reply:
I'm waiting for https://github.com/elastic/kibana/pull/59551 to land and I'll remove